cf-agent/cf-agent.c

```diff
@@ -238,11 +238,21 @@ int main(int argc, char *argv[])
     {
         Log(LOG_LEVEL_ERR, "CFEngine was not able to get confirmation of promises from cf-promises, so going to failsafe");
         EvalContextClassPutHard(ctx, "failsafe_fallback", "attribute_name=Errors,source=agent");
-        GenericAgentConfigSetInputFile(config, GetInputDir(), "failsafe.cf");
-        policy = LoadPolicy(ctx, config);
+        
+        if (CheckAndGenerateFailsafe(GetInputDir(), "failsafe.cf"))
+        {
+            GenericAgentConfigSetInputFile(config, GetInputDir(), "failsafe.cf");
+            policy = LoadPolicy(ctx, config);
+        }
     }
     assert(policy);
-
+    
+    if (!policy)
```

Comment:
Having both this check and the `assert()` is redundant. I'd say that the `assert()` is all you need, since `LoadPolicy()` can never return NULL, and it is a programming error if it does.